dashboard/src/components/ProvisionButton/index.tsx

```diff
@@ -46,34 +39,40 @@ const RequiredRBACRoles: IRBACRole[] = [
   },
 ];
 
+const NameProperty: JSONSchema6 = {
+  description: "Name for ServiceInstance",
+  type: "string",
+};
+
 class ProvisionButton extends React.Component<IProvisionButtonProps, IProvisionButtonState> {
   public state: IProvisionButtonState = {
     isProvisioning: false,
     modalIsOpen: false,
-    parameters: JSON.stringify(
-      {
-        firewallRules: [
-          {
-            endIPAddress: "255.255.255.255",
-            name: "AllowAll",
-            startIPAddress: "0.0.0.0",
-          },
-        ],
-        location: "eastus",
-        resourceGroup: "default",
-        sslEnforcement: "disabled",
-      },
-      undefined,
-      2,
-    ),
-    releaseName: "",
-    selectedClass: this.props.selectedClass,
-    selectedPlan: this.props.selectedPlan,
   };
 
   public render() {
-    const { plans, classes } = this.props;
-    const { selectedClass, selectedPlan } = this.state;
+    const { selectedPlan } = this.props;
+    let schema = selectedPlan.spec.instanceCreateParameterSchema;
+    if (schema) {
+      schema.properties = {
+        name: NameProperty,
+        ...(schema.properties || {}),
+      };
+      schema.required = [...(schema.required || []), "name"];
```

Comment:
where are the required fields validated?

Reply:
For now this just uses the basic browser validations with the `required="true"` attribute on input elements.